### DISCUSSION TITLE
Redirect node naming

### DISCUSSION THREAD

**Discussion by ghost** [IndiePython/nodezator#84] [Feature Request]
Are redirect nodes purposefully limited to the input's name? Many times I'd rename them to something more useful in a context.

E: they can be renamed until an input is connected.

**Comment by KennedyRichard** [comment]
I think it is not the first time people have requested this behaviour. The reason I didn't consider it for implementation yet is because I wanted to give it more thought before deciding on it, because, as you know by now, the generalist nature of Nodezator require us to carefully think about our design decisions.

However, after thinking about it for awhile, it actually dawned on me that this is actually a pretty basic/fundamental feature of any programming language. That is, the ability to rename a variable. Since Nodezator aims to be a visual representation of Python as faithfully as possible, it is only natural that renaming should be possible as well.

Regardless of my opinion, I still think this is not as urgent or important as other features. For instance, we have text block objects that we can use as notes to provide additional context. Also, it will likely require some non-trivial changes in both graph execution and Python exporting.

Nonetheless, the change seems worth the cost. The ability to rename data travelling the graph may indeed improve understanding and readability of the graph and, as I said before, is something that comes naturally with programming languages, not only Python.

I'll upgrade the category of this discussion to "feature request" for now, since I think the suggestion is worth seriously considering for implementation in the future.

Users in the past have also requested the ability to change the name of outputs of individual node instances, for the same reason, that is, to provide extra context to the outcoming data, but I think that just providing a way to rename data with redirect nodes would be enough to satisfy both demands.

The only additional change that I'd make for the feature would be to include some kind of visual cue to make it evident that the redirect node was being used to rename data rather than redirecting it. I'd actually call them "Rename nodes" instead, and they'd have some icon or other different visual to make it obvious it is renaming the data going through it.

**Reply by ghost** [reply]
Wouldn't it be possible to just make it a visual change (and perhaps show the original name too)?

(Currently I use text nodes for added context too, so it's not that urgent as you mentioned.)
